ui/lib/replication/addon/components/replication-summary.js

```diff
@@ -66,13 +67,15 @@ export default Component.extend(ReplicationActions, DEFAULTS, {
     this.setProperties(DEFAULTS);
   },
 
-  submit: task(function* () {
-    try {
-      yield this.submitHandler.perform(...arguments);
-    } catch (e) {
-      // do not handle error
-    }
-  }),
+  submit: task(
+    waitFor(function* () {
```

Comment:
oh, interesting -- normally i'd wouldn't expect to see things from `@ember/test-waiters` to be used in production code. usually i'd expect ember concurrency to handle the async behavior and then using `waitFor` or `waitUntil` inside the tests to ensure async tasks finish before making assertions. i see that that's used in a few places around the app though, so seems like an intentional decision. was this necessary here to resolve the flaky tests?

and as a side note (does not have to be answered in this PR, but more for me own learning) do you have any background context on how this pattern was adopted?

Reply:
Good question! The repo docs are awful/non-existent but on the [method description here](https://github.com/emberjs/ember-test-waiters/blob/fefea6eecd20e13fc20f7757564889b4bc597c00/addon/addon/wait-for.ts#L16) it describes the usage: 

> A convenient utility function to simplify waiting for async. Can be used in both decorator and function form. When applied to an async function, it will cause tests to wait until the returned promise has resolves. When applied to a generator function, it will cause tests to wait until the returned iterator has run to completion, which is useful for wrapping ember-concurrency task functions.

We first started using it in this [ember upgrade](https://github.com/hashicorp/vault/pull/13443), where we were seeing a lot of flakiness due to promises not resolving. This helper helps stabilize those tests without adding custom waiters everywhere, and stabilizing flakiness is also why I added it here. Hope that helps!